### PR TITLE
{data}[foss/2022b] Embree v3.13.2, HDF5 v1.10.3, MEDCoupling v9.10.0, ... w/ Python 3.10.8

### PR DESCRIPTION
**Pull request by Iceman9**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by smoors** [comment]
@Iceman9 thank you for your contribution!

we are quite strict about the format of easyconfigs, so please don't be offended by the remarks :)

File: easybuild/easyconfigs/s/SALOME/SALOME_CONFIGURATION-9.10.0.eb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+sources = [{
+    "filename": 'CONFIGURATION-%(version)s.tar.gz',
+    "git_config": {
+        "url": "http://git.salome-platform.org/gitpub/tools",
+        "repo_name": "configuration",
+        "tag": "V9_10_0",
+        "recursive": True
+    }
+}
+]
```
Comment (smoors) [inline easybuild/easyconfigs/s/SALOME/SALOME_CONFIGURATION-9.10.0.eb:18]:
```suggestion
}]
```

File: easybuild/easyconfigs/s/SALOME/SALOME_CONFIGURATION-9.10.0.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    'files': [],
+    'dirs': ['cmake', 'autotools'],
+}
+
+extract_sources = True
+
+toolchain = {
+    'name': 'system',
+    'version': ''
+}
```
Comment (smoors) [inline easybuild/easyconfigs/s/SALOME/SALOME_CONFIGURATION-9.10.0.eb:30]:
```suggestion
toolchain = SYSTEM
```

File: easybuild/easyconfigs/s/SALOME/SALOME_CONFIGURATION-9.10.0.eb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name = "SALOME_CONFIGURATION"
+easyblock = "Binary"
+version = "9.10.0"
+
+homepage = 'https://www.salome-platform.org'
+description = """Configuration files and other utilities for SALOME
+platform."""
+
+sources = [{
+    "filename": 'CONFIGURATION-%(version)s.tar.gz',
```
Comment (smoors) [inline easybuild/easyconfigs/s/SALOME/SALOME_CONFIGURATION-9.10.0.eb:10]:
```suggestion
    "filename": SOURCE_TAR_GZ,
```

File: easybuild/easyconfigs/m/MEDCOUPLING/MEDCOUPLING-9.10.0-foss-2022b.eb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+homepage = 'https://www.salome-platform.org'
+description = """The MEDCoupling tool gathers several powerful functionalities
+around the input and output data of simulation codes (meshes and fields
+mainly)."""
+
+source_urls = ['https://files.salome-platform.org/Salome/other']
+
+
+sources = [{
+    "filename": 'medcoupling-%(version)s.tar.gz',
```
Comment (smoors) [inline easybuild/easyconfigs/m/MEDCOUPLING/MEDCOUPLING-9.10.0-foss-2022b.eb:14]:
```suggestion
    "filename": SOURCE_TAR_GZ,
```

File: easybuild/easyconfigs/m/MEDCOUPLING/MEDCOUPLING-9.10.0-foss-2022b.eb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+version = "9.10.0"
+
+homepage = 'https://www.salome-platform.org'
+description = """The MEDCoupling tool gathers several powerful functionalities
+around the input and output data of simulation codes (meshes and fields
+mainly)."""
+
+source_urls = ['https://files.salome-platform.org/Salome/other']
+
+
```
Comment (smoors) [inline easybuild/easyconfigs/m/MEDCOUPLING/MEDCOUPLING-9.10.0-foss-2022b.eb:12]: only 1 line

File: easybuild/easyconfigs/m/MEDCOUPLING/MEDCOUPLING-9.10.0-foss-2022b.eb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+sources = [{
+    "filename": 'medcoupling-%(version)s.tar.gz',
+    "git_config": {
+        "url": "http://git.salome-platform.org/gitpub/tools",
+        "repo_name": "medcoupling",
+        "tag": "V9_10_0",
+        "recursive": True
+    }
+},
+]
```
Comment (smoors) [inline easybuild/easyconfigs/m/MEDCOUPLING/MEDCOUPLING-9.10.0-foss-2022b.eb:22]:
```suggestion
}]
```

**Review by smoors — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/e/Embree/Embree-3.13.2-foss-2022b.eb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    'name': 'foss',
+    'version': '2022b'
+}
+
+toolchainopts = {'pic': True}
+
+
+dependencies = [
+    ('ispc', '1.15.0', '', SYSTEM),
+    ('tbb', '2020.3'),
```
[the remark this answers is not in the document]
Comment (Iceman9) [reply easybuild/easyconfigs/e/Embree/Embree-3.13.2-foss-2022b.eb:21]: I haven't tested up to which newer TBB version is compatible with this Embree version. When Intel released the OneAPI toolbox I haven't checked how much does the API change, i.e., if I can simply switch to the new version or provide patches... Additionally the software I am working on uses this version with Embree and not the newer one. 

tl;dr sadly not possible.

**Comment by Iceman9** [comment]
@smoors I should've read the ``eb`` command options with more care. At this point I can do a rebase instead of having these commits inside.

Regarding the use of older versions for the Embree, HDF5 and TBB packages, I haven't tested what happens if I directly switch to newer versions. The versions are verbatim to what is used in the SALOME 9.10.0 software framework.

File: easybuild/easyconfigs/e/Embree/Embree-3.13.2-foss-2022b.eb (hunk 7 above)
[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/e/Embree/Embree-3.13.2-foss-2022b.eb:21]: for getting an easyconfig accepted in the EB repo, we have a rule saying that for each toolchain generation there should only be 1 version of a software used as a dependency for other software. in the case of `2022b`, the latest generation, we should really try to use the lastest `tbb` version if possible.

can you test this? the chances are high that it is backwards compatible. if not, i suggest using the `2021b` generation instead.

[the remark this answers is not in the document]
Comment (Iceman9) [reply easybuild/easyconfigs/e/Embree/Embree-3.13.2-foss-2022b.eb:21]: I understand, then I will close the PR. The software is frozen in terms of versions, and I do not have time to see if upgrading to latest Embree/Tbb (now I see Embree is >4.0.0) or to the version that works with the 2022b toolchain is compatible with all the software packages in SALOME. 

The point of this PR was to have the dependencies of a program I am working on (I have a EC but for a different repository) and then to gradually include all the EasyConfigs of the SALOME framework, so that it is not installed as a Binary EasyConfig. I also intend to always use the newest toolchain but given the policy this will just make a big mess.
